src/lib/hooks/useUniqueUsernameValidation/index.ts

```diff
@@ -7,7 +7,7 @@ const checkIfUsernameIsUnique = async (username?: string): Promise<boolean> => {
   const { data: usersWithMatchingUsernames, error } = await supabase
     .from<definitions["user_profiles"]>("user_profiles")
     .select("id")
-    .ilike("name", `%${username}%`);
+    .ilike("name", `%${username.trim()}%`);
```

Comment:
Why is the `trim` introduced here?

Reply:
This makes it defensive against leading and trailing spaces in the usernames. In a perfect world, this should not be necessary but with this additional step, no problem of leading/trailing slashes should occur.